### DISCUSSION TITLE
Supported file extensions for (regular) uploads

### DISCUSSION THREAD

**Discussion by mikekasprzak** [ludumdare/ludumdare#2171] [Ideas]
At the time of this writing we support only the following file formats (extensions) when uploading.

  * Archives: `zip`, `tar.gz`, `tar.bz2`, `7z`, `rar`
  * Microsoft: `msi`, `exe`, `com`
  * Apple: `dmg`, `ipa`
  * Linux: `flatpak`, `snap`, `appimage`, `deb`, `rpm`, `pkg`, `apk` (Android)
  * Document: `pdf`, `epub`, `txt`
  * Legacy: `jar`, `swf`

Let us know if there's something we should add.

**Comment by local-minimum** [comment]
How about the `.love` and `.jar`, maybe `.py` for those that make singlefile games? I feel like att least the former one usually encounters (could be misremembering endings exactly). I have a feeling there are some more that happen from time to time.  Pdf should be there too for those that make printable games and probably some more document formats.

**Reply by mikekasprzak** [reply]
`.py` is a source file (you might be thinking `.pyc`). I'm not sure what a `.love` file contains.

Good idea `.pdf` and good idea `.jar`, and on that note `.apk` (Android). Both are `zip` files in disguise.

**Reply by local-minimum** [reply]
`.py` is source but quite runnable on linux with just a shebang at the top

**Reply by mikekasprzak** [reply]
Any file can be run if it has a _shebang_. :wink:

I'm concerned about a couple issues:
* someone could upload every `.py` file in their project, before realizing we don't allow `.png`, `.json`, etc.
* naturally we'd have to support `.sh`, `.bat`, and whatever powershell uses. This in turn leads to this thread blowing up with a request and backlog of every esoteric script format imaginable (`lua`, `nut`, `php`, etc). At that point, we may as well open the file extension floodgates.

I don't want to, for example, allow image file formats. An new user could find themself using the file uploader when they should be using the image uploader.

**Reply by mikekasprzak** [reply]
I've updated the lists.

**Comment by Admicos** [comment]
For linux, [`.AppImage`](https://appimage.org/) could be useful, if anybody actually uses it.

**Reply by mikekasprzak** [reply]
Fair. For some reason I do have `.AppImage` on the list in the code, but it's commented out. I can't quite figure out why. :thinking:
